backend/src/main/java/org/ukma/spring/crooodle/service/impl/HotelServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package org.ukma.spring.crooodle.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.ukma.spring.crooodle.dto.*;
+import org.ukma.spring.crooodle.dto.common.PageResponseDto;
+import org.ukma.spring.crooodle.entities.HotelEntity;
+import org.ukma.spring.crooodle.repository.CountryRepository;
+import org.ukma.spring.crooodle.repository.WorldRegionRepository;
+import org.ukma.spring.crooodle.repository.HotelRepository;
+import org.ukma.spring.crooodle.service.HotelService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Service
+public class HotelServiceImpl implements HotelService {
+
+    private final HotelRepository hotelRepository;
+    private final CountryRepository countryRepository;
+    private final WorldRegionRepository regionRepository;
+
+    @Override
+    public Long create(HotelCreateRequestDto requestDto) {
+        var country = countryRepository.findById(requestDto.getCountryId())
+            .orElseThrow(() -> new IllegalArgumentException("Country not found"));
+
+        var region = requestDto.getRegionId() != null
+            ? regionRepository.findById(requestDto.getRegionId())
+            .orElseThrow(() -> new IllegalArgumentException("Region not found"))
+            : null;
+
+        HotelEntity hotel = HotelEntity.builder()
+            .name(requestDto.getName())
+            .address(requestDto.getAddress())
+            .country(country)
+            .country_region(region)
+            .build();
+
+        hotel = hotelRepository.save(hotel);
+        return hotel.getId();
+    }
+
+    @Override
+    public HotelAdminResponseDto getAdmin(Long id) {
+        HotelEntity hotel = hotelRepository.findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("Hotel not found"));
+
+        return mapToAdminDto(hotel);
+    }
+
+    @Override
+    public HotelResponseDto get(Long id) {
+        HotelEntity hotel = hotelRepository.findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("Hotel not found"));
+
+        return mapToDto(hotel);
+    }
+
+    @Override
+    public PageResponseDto<HotelAdminResponseDto> listAdmin(HotelCriteriaDto criteriaDto) {
+        List<HotelEntity> hotels = hotelRepository.findAll();
```

Comment:
Does not use criteria in any way? Is this your intended behaviour? Than at least put TODO